### ISSUE TITLE
Add more Layers

### ISSUE THREAD

**Issue by cgarciae** [cgarciae/treex#40]
The idea is to add more wrappers over Flax layers.

## Guidelines
* Argument names and types should be the same unless there is a good reason to deviate from Flax API.
* "params" and "batch_stats" are assigned to fields, take implementation of [Linear](https://github.com/cgarciae/treex/blob/master/treex/nn/linear.py) and [BatchNorm](https://github.com/cgarciae/treex/blob/master/treex/nn/batch_norm.py) as references.
* Must take into account `training` and `frozen` state, checkout the implementation of [BatchNorm](https://github.com/cgarciae/treex/blob/master/treex/nn/batch_norm.py) and [Dropout](https://github.com/cgarciae/treex/blob/master/treex/nn/dropout.py) as references. 
* Must add test for numerical equivalence, check out [test_linear.py](https://github.com/cgarciae/treex/blob/689b8a24347c50f24d69e546b384c81a59ebdab0/tests/nn/test_linear.py#L46) as a reference.

### Layers List
- [ ] [ConvTranspose](https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/linear.py#L302)
- [ ] [Embed](https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/linear.py#L386) - should we call it `Embedding`?
- [ ] [LayerNorm](https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/normalization.py#L224)
- [ ] [GroupNorm](https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/normalization.py#L273)
- [ ] [MaxPooling](https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/pooling.py#L91) - Module version
- [ ] [AvgPooling](https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/pooling.py#L72) - Module version
- [ ] [LSTM](https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/recurrent.py#L66) - Follow Keras API, implement it using `scan` but add a flag for unrolling it.
- [ ] [GRU](https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/recurrent.py#L279) - Follow Keras API, implement it using `scan` but add a flag for unrolling it.

If you want to work on a layer, feel free to "claim" it via a comment here so other know you are working on it.

**Comment by ptigwe** [comment]
On the embedding layer, I think it'll be fine with `Embed` as opposed to `Embedding` primarily due to the Flax-like design, plus it seems to be the name used in especially given the first point you make on the guidelines. At the same time, I'm not opposed to `Embedding` as its used in most frameworks. To the best of my knowledge of the other JAX-based NN libraries, only Trax calls it `Embedding`, Haiku and Flax both call it `Embed`.

**Comment by cgarciae** [comment]
@ptigwe Agreed. I'd leave it up to the implementer to decide the name.
